src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -35,148 +36,160 @@
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Inject;
+import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.util.Objects.requireNonNull;
 
 public class SlackEventNotification implements EventNotification {
 
-	public interface Factory extends EventNotification.Factory {
-		@Override
+    private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+    private final EventNotificationService notificationCallbackService;
+    private final Engine templateEngine;
+    private final NotificationService notificationService;
+    private final ObjectMapper objectMapper;
+    private final NodeId nodeId;
+    private final SlackClient slackClient;
+
+    @Inject
+    public SlackEventNotification(EventNotificationService notificationCallbackService,
+                                  ObjectMapper objectMapper,
+                                  Engine templateEngine,
+                                  NotificationService notificationService,
+                                  NodeId nodeId, SlackClient slackClient) {
+        this.notificationCallbackService = notificationCallbackService;
+        this.objectMapper = requireNonNull(objectMapper);
+        this.templateEngine = requireNonNull(templateEngine);
+        this.notificationService = requireNonNull(notificationService);
+        this.nodeId = requireNonNull(nodeId);
+        this.slackClient = requireNonNull(slackClient);
+    }
+
+    /**
+     * @param ctx
+     * @throws PermanentEventNotificationException is thrown with bad webhook url, authentication error type issues
+     * @throws TemporaryEventNotificationException is thrown for network or timeout type issues
+     */
+    @Override
+    public void execute(EventNotificationContext ctx) throws PermanentEventNotificationException, TemporaryEventNotificationException {
+        final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+        ValidationResult result = config.validate();
+        result.getErrors().entrySet().stream().forEach(e -> LOG.error("Invalid configuration for key [{}] and value [{}]", e.getKey(), e.getValue()));
+
+        LOG.debug("SlackEventNotification backlog size in method execute is [{}]", config.backlogSize());
+
+        if (result.failed()) {
+            throw new PermanentEventNotificationException("Please verify your Slack Event Configuration");
+        }
```

Comment:
As mentioned previously, I don't think there's any reason to call config.validate() here. The validation happens when a user creates the new Notification via the UI. This should be removed.
```suggestion
        LOG.debug("SlackEventNotification backlog size in method execute is [{}]", config.backlogSize());
```

Reply:
Thanks for your feedback. I will perform some more testing locally, before I commit this suggestion.

Reply:
Not a blocker, but I'm going to keep bringing it up every time this code is up for review.